File: app/src/test/java/org/imozerov/streetartview/TestStreetArtViewApp.java
package org.imozerov.streetartview;

/**
 * Created by imozerov on 04.03.16.
 */
public class TestStreetArtViewApp extends StreetArtViewApp {
    private AppComponent appComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        appComponent = DaggerAppComponent.builder()
                .appModule(new TestAppModule(this))
                .build();
    }

    @Override
    public AppComponent getAppComponent() {
        return appComponent;
    }
}

File: app/src/test/java/org/imozerov/streetartview/TestStreetArtViewApp.kt
package org.imozerov.streetartview

import org.imozerov.streetartview.network.NetModule

/**
 * Created by imozerov on 04.03.16.
 */
class TestStreetArtViewApp : StreetArtViewApp() {
    override var appComponent: AppComponent? = null
    private val mockUrl = "https://dummy.com"

    override fun onCreate() {
        super.onCreate()
        appComponent = DaggerAppComponent.builder()
                .appModule(TestAppModule(this))
                .netModule(NetModule(mockUrl))
                .build()
    }
}
